### ISSUE TITLE
[Specification question] Can an ESYS_TR really be NULL?

### ISSUE THREAD

**Issue by Superhepper** [tpm2-software/tpm2-tss#2357]
tpm2-tss version: 3.1.0

I was trying to do a little bit of debugging of our rust wrapper that when running Valgrind complained a lot about ```Conditional jump or move depends on uninitialised value``` inside marshalling functions such as, but not limited to, ```Tss2_MU_TPM2B_NONCE_Marshal``` when a call to ```Esys_StartAuthSession``` was made. So I went to the specification to try to see if there was something that we had missed.

I then found the following which confused me:
*TSS Enhanced System API (ESAPI) Specification 1.00 Revision 14, page 50*
>tpmKey: Can be NULL if no salting is required (this represents the case of TPM_RH_NULL from
>the TPM’s library specification).
>• bind: Can be NULL (this represents the case of TPM_RH_NULL from the TPM’s library
>specification).

This is quite confusing because ```tpmKey``` and ```bind``` are ESYS_TR and I thought they were supposed to be set to ESYS_TR_NONE when they are not set and not to NULL which I interpret should mean 0?

**Comment by williamcroberts** [comment]
Yeah thats a typo in the spec it should state it can be `ESYS_TR_NONE`.

**Comment by Superhepper** [comment]
Don't forget this one while writing the errata.

https://github.com/tpm2-software/tpm2-tss/issues/1930
